Modules/_sqlite/connection.c

```diff
@@ -827,18 +827,17 @@ PyObject* pysqlite_connection_create_function(pysqlite_Connection* self, PyObjec
         return NULL;
     }
 
+    if (PyDict_SetItem(self->function_pinboard, func, Py_None) == -1) {
+        return NULL;
+    }
     rc = sqlite3_create_function(self->db, name, narg, SQLITE_UTF8, (void*)func, _pysqlite_func_callback, NULL, NULL);
-
     if (rc != SQLITE_OK) {
+        PyDict_DelItem(self->function_pinboard, func);
```

Comment:
Why do we bother to delete the function from the dict? If call pysqlite_connection_create_function() twice with the same Python function, and the second call fails, the reference to the function will be deleted, and this can cause a crash when use the first Sqlite function.

Reply:
Makes sense.